packages/agreement-process/src/services/agreementService.ts

```diff
@@ -1145,6 +1188,62 @@ export function agreementServiceBuilder(
         await repository.createEvent(event);
       }
     },
+    async verifyTenantCertifiedAttributes(
+      {
+        tenantId,
+        descriptorId,
+        eserviceId,
+      }: {
+        tenantId: TenantId;
+        descriptorId: DescriptorId;
+        eserviceId: EServiceId;
+      },
+      { logger, authData }: WithLogger<AppContext>
+    ): Promise<agreementApi.HasCertifiedAttributes> {
+      logger.info(
+        `Veryfing tenant ${tenantId} has required certified attributes for descriptor ${descriptorId} of eservice ${eserviceId}`
+      );
+
+      const isSameOrganization = authData.organizationId === tenantId;
+
+      const hasValidDelegation = isSameOrganization
+        ? await isConsumerDelegationByEserviceAndDelegatorIdsValid(
+            {
+              eserviceId,
+              delegatorId: authData.organizationId,
+            },
+            readModelService
+          )
+        : await isConsumerDelegationByEserviceAndIdsValid(
+            {
+              eserviceId,
+              delegateId: authData.organizationId,
+              delegatorId: tenantId,
+            },
+            readModelService
+          );
```

Comment:
There should be no need to query delegations if the requester has the same tenant ID specified in the body: it means they want to consume the service for themselves. We should query delegations only if the specified tenant is different, because the requester wants to consumer a service for another tenant (so, we check that the corresponding delegation actually exists and is active)

Reply:
> There should be no need to query delegations if the requester has the same tenant ID specified in the body: it means they want to consume the service for themselves. We should query delegations only if the specified tenant is different, because the requester wants to consumer a service for another tenant (so, we check that the corresponding delegation actually exists and is active)

I think that check is needed because [In the SRS](https://pagopa.atlassian.net/wiki/spaces/PDNDI/pages/1236959344/SRS+Incaricato#Creazione-richiesta-di-fruizione) is specified that the delegator CANNOT create an agreement if it has an already active delegation on the eservice; so even, if the requester === tenandId we have to check that the requester isn't a delegator for that eservice

Reply:
You're right 🚀  As discussed yesterday, if possible I would try to make this code more readable, maybe taking inspiration from the similar check that the catalog process performs for producer delegations: https://github.com/pagopa/interop-be-monorepo/blob/dac1a6c47564c8c7efad1c1637928fe7586df523/packages/catalog-process/src/services/validators.ts#L35

Still, it's non blocking, I'm approving the PR ✅